python/Ganga/GPIDev/Lib/GangaList/GangaList.py

```diff
@@ -421,7 +434,10 @@ def __str__(self):
         return self.toString()
 
     def append(self, obj, my_filter=True):
-        self._list.append(self.strip_proxy(obj, my_filter))
+        elem = self.strip_proxy(obj, my_filter)
+        if isType(elem, GangaObject):
+            stripProxy(elem)._setParent(stripProxy(self)._getParent())
```

Comment:
It should never be required to write `stripProxy(self)` as we are inside a class which is non a proxy class. If you find a situation where `self` is a proxy object then that is a major thing that needs fixing.